src/map-marker/fixed-marker.js

```diff
@@ -92,18 +102,32 @@ const FixedMarker = ({
   color = color || primaryB;
   background = background || backgroundInversePrimary;
 
-  const doesPinHeadTransformOnDrag = needle !== NEEDLE_SIZES.none;
+  const doesPinHeadTransformOnDrag =
+    needle !== NEEDLE_SIZES.none &&
+    size !== PINHEAD_SIZES_SHAPES.xxSmallCircle &&
+    size !== PINHEAD_SIZES_SHAPES.xxSmallSquare;
 
   const [Root, rootProps] = getOverrides(overrides.Root, StyledRoot);
   const [
     FixedMarkerDragContainer,
     fixedMarkerDragContainerProps,
   ] = getOverrides(overrides.DragContainer, StyledFixedMarkerDragContainer);
+
+  const renderNeedle =
+    needle !== NEEDLE_SIZES.none &&
+    size !== PINHEAD_SIZES_SHAPES.xxSmallCircle &&
+    size !== PINHEAD_SIZES_SHAPES.xxSmallSquare;
+  if (!renderNeedle) {
+    console.warn(
+      `Needles cannot be rendered with ${PINHEAD_SIZES_SHAPES.xxSmallCircle} or ${PINHEAD_SIZES_SHAPES.xxSmallSquare} pin heads`,
+    );
+  }
+
   return (
-    <Root data-baseweb="fixed-map-marker" {...rootProps}>
+    <Root data-baseweb="fixed-map-marker" {...restProps} {...rootProps}>
```

Comment:
We usually don't spread rest props on the root element because it's available through overrides. Is there a compelling reason to do so?

Reply:
I've added that to be able to pass `onMouseEnter`, `onMouseLeave`, `onClick` event handlers. Passing them through overrides is more verbose.. I guess for these specific properties I can explicitly pass them to the `Root`. Would that be a better approach?